### ISSUE TITLE
Arabic Language

### ISSUE THREAD

**Issue by Mo9a7i** [simov/slugify#100]
Hello,

How can slugify transform this:
`السلام عليكم ورحمة الله وبركاته`

Into this:
`السلام-عليكم-ورحمة-الله-وبركاته`

shall I add the Arabic characters to `charmap.json` as below ?
```json
{
"أ": "",
"ب": "ب",
"ت": "ت",
"ث": "ث",
"ج": "ج",
"ح": "؛"
}
```
and so on ?

**Comment by Mo9a7i** [comment]
Currently, this:
`السلام عليكم ورحمة الله وبركاته`

slugifies into:
``

nothing.

**Comment by simov** [comment]
Yes if you see characters missing add them to the `charmap.json` file :+1:

**Comment by Mo9a7i** [comment]
Didn't work @simov seems like it is the same issue as here #89 
maybe there should be a location where allowed characters can be added.

**Comment by simov** [comment]
You have to specify the transliteration using latin characters, for example:

```
"Ș": "S",
```

and in your case:

```
"ث": "t",
```

or whatever the phonetic (sound) actually is.

**Comment by Mo9a7i** [comment]
The need is for the slog to be as such:
`السلام-عليكم-ورحمة-الله-وبركاته`

example:
<https://www.f6sny.com/jokes/اثنين-وقفوا-لهم-سيارة-تاكسي-واحد-ركب-والثاني-سيقان>

**Comment by simov** [comment]
In this case you have to use the `extend` method and define your mappings there. Not sure if your case have to be integrated into this module.

**Comment by ZaidBarghouthi** [comment]
@simov 
How about an option to allow utf symbols and only convert preserved chars.

I'm currently facing the same issue as @Mo9a7i, as we like to keep Arabic text in the slug in many Arabic only pages.

**Comment by simov** [comment]
Ok, now I understand, I think this can be implemented as an option. Probably `charmap: false` to opt out of the default behavior.

**Comment by simov** [comment]
I was about to start working on this feature but I noticed that the issue comes from the default `remove` regex:

```js
/[^\w\s$*_+~.()'"!\-:@]+/g
```

This regex removes anything that's not in that list, meaning it preserves only the listed characters. The `\w` flag stands for the `a-zA-Z` range, meaning anything except those characters will be removed.

So instead we can update the `remove` regex to remove only the listed characters:

```js
slugify('السلام عليكم ورحمة الله وبركاته', {
  remove: /[$*_+~.()'"!\-:@]+/g
})
```

And this yields the expected result:

```
السلام-عليكم-ورحمة-الله-وبركاته
```

**Comment by ZaidBarghouthi** [comment]
@simov yup, that would word.

**Comment by the94air** [comment]
For easy copy 👍 
```js
const arabic = {
  "ء" : "ء",
  "ا" : "ا",
  "أ" : "أ",
  "إ" : "إ",
  "آ" : "آ",
  "ؤ" : "ؤ",
  "ئ" : "ئ",
  "ب" : "ب",
  "ت" : "ت",
  "ث" : "ث",
  "ج" : "ج",
  "ح" : "ح",
  "خ" : "خ",
  "د" : "د",
  "ذ" : "ذ",
  "ر" : "ر",
  "ز" : "ز",
  "س" : "س",
  "ش" : "ش",
  "ص" : "ص",
  "ض" : "ض",
  "ط" : "ط",
  "ظ" : "ظ",
  "ع" : "ع",
  "غ" : "غ",
  "ف" : "ف",
  "ق" : "ق",
  "ك" : "ك",
  "ل" : "ل",
  "م" : "م",
  "ن" : "ن",
  "ه" : "ه",
  "و" : "و",
  "ي" : "ي",
  "ة" : "ة",
  "ى" : "ى",
};
```

**Comment by Trott** [comment]
If you're not trying to get a Latin alphabet result and if you know that your input strings will use a single alphabet, I imagine using `String.prototype.replace()` might be easier (and have a smaller code footprint) than using `slug()`.

```javascript
const regex = /[^\d\u0621-\u064A]+/g;
const string = 'السلام عليكم ورحمة الله وبركاته';

console.log(string.replace(regex, '-')); // 'السلام-عليكم-ورحمة-الله-وبركاته'
```

**Comment by alieldeba** [comment]
Thanks so much i was facing this issue  ❤️
